// nhentai/lib/doujin.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var image_1 = require("./image");
var tag_1 = require("./tag");
var api_1 = require("./api");
var Doujin = /** @class */ (function () {
    function Doujin(book, api) {
        var _this = this;
        this.id = book.id;
        this.mediaId = +book.media_id;
        this.titles = book.title;
        this.scanlator = book.scanlator;
        this.uploadDate = new Date(book.upload_date * 1000);
        this.length = book.num_pages;
        this.favorites = book.num_favorites;
        this.url = api_1.HOST_URL + "/g/" + book.id;
        this.pages = book.images.pages.map(function (image, index) { return new image_1.default(image, index + 1, _this); });
        this.cover = new image_1.default(book.images.cover, 'cover', this);
        this.thumbnail = new image_1.default(book.images.thumbnail, 'thumbnail', this);
        this.tags = book.tags.map(function (tag) { return new tag_1.default(tag); });
        if (api.options.preserveRaw)
            this.raw = book;
    }
    Doujin.prototype.hasTagByName = function (name) {
        return !!this.tags.find(function (tag) { return tag.name === name; });
    };
    Doujin.prototype.hasTagByID = function (ID) {
        return !!this.tags.find(function (tag) { return tag.id === ID; });
    };
    return Doujin;
}());
exports.default = Doujin;
