// nhentai/lib/image.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var node_fetch_1 = require("node-fetch");
var api_1 = require("./api");
var Image = /** @class */ (function () {
    function Image(image, name, doujin) {
        /**
         * Page number if this is a page. If it is not, then its null and either a cover or thumbnail
         */
        this.page_number = null;
        this.extension = Image.extensionConvert(image.t);
        this.height = image.h;
        this.width = image.w;
        var parsedName = Number(name);
        this.url = (isNaN(parsedName) ? api_1.THUMBS_URL : api_1.IMAGE_URL) + "/galleries/" + doujin.mediaId + "/" + name + "." + this.extension;
    }
    /**
     * Fetches the image
     */
    Image.prototype.fetch = function () {
        var _this = this;
        return new Promise(function (resolve, reject) {
            node_fetch_1.default(_this.url)
                .then(function (data) { return data.buffer(); })
                .then(function (data) { return resolve(data); })
                .catch(function (error) { return reject(error); });
        });
    };
    /**
     * Converts an images `t` paramater to a file extension\
     * Only 3 are known so far
     * @param extension Raw type from the api
     */
    Image.extensionConvert = function (extension) {
        switch (extension) {
            case 'p':
                return 'png';
            case 'j':
                return 'jpg';
            case 'g':
                return 'gif';
            default:
                throw new Error("Image extension \"" + extension + "\" is not a known format.");
        }
    };
    return Image;
}());
exports.default = Image;
